tools/add_triage_signature.py

```diff
@@ -2399,6 +2399,72 @@ def _process_ticket(self, url, issue_key):
             )
 
 
+class ControllerFailedToStart(ErrorSignature):
+    """
+    This signature looks for missing pivot URL in rpm-ostree status of the control-plane hosts
+    """
+
+    PODS_PATH = "*_bootstrap_*.tar.gz/logs_host_*/log-bundle-*.tar.gz/log-bundle-*/resources/pods.json"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(
+            *args,
+            **kwargs,
+            function_impact_label="assisted_controller_failed_to_start",
+            comment_identifying_string="h1. Assisted Installer Controller failed to start",
+        )
+
+    def _process_ticket(self, url, issue_key):
+        cluster = get_metadata_json(url)["cluster"]
+        bootstrap_node = [host for host in cluster["hosts"] if host["bootstrap"]][0]
+
+        if bootstrap_node["progress"]["current_stage"] != "Waiting for controller":
+            return
+
+        triage_logs_tar = get_triage_logs_tar(triage_url=url, cluster_id=cluster["id"])
+        try:
+            # Fetch pods.json from the bootstrap log bundle
+            pods_json = triage_logs_tar.get(self.PODS_PATH)
+        except FileNotFoundError:
+            return
+
+        # Fetch assisted-controller status
+        try:
+            pods = json.loads(pods_json)
+            try:
+                controller_pod = [
+                    pod
+                    for pod in pods.get("items", [])
+                    if pod.get("metadata", {}).get("namespace") == "assisted-installer"
+                ][0]
+            except IndexError:
+                raise "Failed to find controller pod in pods json"
+            try:
+                ready = [
+                    condition.get("status") == "True"
+                    for condition in controller_pod.get("status", {}).get("conditions", {})
+                    if condition.get("type") == "Ready"
+                ][0]
+            except IndexError:
+                raise f"Failed to determine controller pod in ready condition, here is it's status: {controller_pod['status']}"
+            if not ready:
```

Comment:
would it be beneficial to also output in case of success? (controller is in ready status)

Reply:
:+1:

Reply:
Done